spec/requests/list_all_articles_spec.rb

```diff
@@ -0,0 +1,13 @@
+require 'rails_helper'
+
+describe "Get /api/articles" do
+    let(:category){create(:category)}
+    let!(:articles){ 5.times {FactoryBot.create(:article)}}
+    let(:headers){ {HTTP_ACCEPT: "application/json"} }
+    
+    it "returns a collection of 5 articles" do
+        get '/api/articles', headers: headers
+    
+        expect(JSON.parse(response.body).count).to eq 5
```

Comment:
write a helper file for parsing the response and then you can write this expect block like `expect(response_json.count).to eq 5`

for the helper file 
create a file `spec/support/response_json.rb` and add this in
```
module ResponseJSON
  def response_json
    JSON.parse(response.body)
  end
end

RSpec.configure do |config|
  config.include ResponseJSON
end
```
and add `Dir[Rails.root.join('spec/support/**/*.rb')].each { |f| require f }` to rails_helper.rb

Reply:
done